command/cp.go

```diff
@@ -522,7 +522,12 @@ func (c Copy) doDownload(ctx context.Context, srcurl *url.URL, dsturl *url.URL)
 
 	size, err := srcClient.Get(ctx, srcurl, file, c.concurrency, c.partSize)
 	if err != nil {
-		_ = dstClient.Delete(ctx, dsturl)
+		// file must be closed before deletion
+		file.Close()
```

Comment:
https://github.com/peak/s5cmd/pull/477/files#diff-8e0740ea5af04e2014c49fe30fc0dcb04ccbc849a76160ca4ec80845c248b511R521

will return error because file is first closed here.

Reply:
Yes the "defer file.Close()" will give an error if we close the file here. But that error will be discarded (it won't cause a panic).

As an alternative we can move/write that defer statement to after the [if](https://github.com/Kucukaslan/s5cmd/blob/fix-348-delete-local-file/command/cp.go#L524-L532) statement.

Reply:
Yes the "defer file.Close()" will give an error if we close the file here. But that error will be discarded (it won't cause a panic).

As an alternative we can move/write that defer statement to after the [if](https://github.com/Kucukaslan/s5cmd/blob/fix-348-delete-local-file/command/cp.go#L524-L532) statement.

Reply:
Maybe we can put `defer file.Close()` to else statement to prevent silent error.

Reply:
Yes. I, now, put  it after the if statement. 
The if statements terminates the function, so I didn't enclosed `defer file.Close()` with else.